openml-h2o/src/main/java/com/feedzai/openml/h2o/params/ParametersBuilderUtil.java

```diff
@@ -128,9 +135,75 @@ private ParametersBuilderUtil() { }
                 .filter(modelParam -> !"calibrate_model".equals(modelParam.getName()))
                 // We already pass a seed always whenever possible.
                 .filter(modelParam -> !"seed".equals(modelParam.getName()))
+                // We only want the fields that are present in the list of fields of the algorithm class.
+                .filter(modelParam -> usefulFields.contains(modelParam.getName()))
                 .collect(Collectors.toSet());
     }
 
+    /**
+     * Returns the fields that are set in the given {@linkplain ModelParametersSchemaV3 algorithmClass} as useful for
+     * the algorithm.
+     *
+     * @param algorithmClass The algorithm class to get the useful fields.
+     * @param allParameters  All the parameters to be added if useful fields is not provided by the class.
+     * @return The {@link List} with the useful fields for the given algorithm class.
+     * @since @@@feedzai.next.release@@@
+     */
+    private static List<String> getUsefulFields(final Class<? extends ModelParametersSchemaV3> algorithmClass,
+                                                final Set<String> allParameters) {
+
+        final Field[] declaredFields = algorithmClass.getDeclaredFields();
+        final List<Field> staticFields = new ArrayList<>();
+
+        for (final Field field : declaredFields) {
+            if (isStatic(field.getModifiers())) {
+                staticFields.add(field);
+            }
+        }
+
+        final Optional<Field> usefulFieldsArray = staticFields.stream()
+                .filter(fields -> "fields".equalsIgnoreCase(fields.getName()))
+                .findFirst();
+
+        final ImmutableList.Builder<String> usefulFields = ImmutableList.builder();
+
+        if (usefulFieldsArray.isPresent()) {
+            usefulFields.add(getFieldContent(algorithmClass, usefulFieldsArray.get()));
+        } else {
+            usefulFields.addAll(allParameters);
+        }
+
+        return usefulFields.build();
+    }
+
+    /**
+     * Returns an array with content of the given {@link Field} for the given {@linkplain ModelParametersSchemaV3
+     * algorithmClass}.
+     *
+     * <p> If the given field is not a {@link String} array an empty array is returned.
+     *
+     * @param algorithmClass The class with the given {@code field}.
+     * @param field          The field to retrieve the value.
+     * @return The array with the field content.
+     * @since @@@feedzai.next.release@@@
```

Comment:
Is this annotation valid in this repository? (double-checking)

Reply:
It doesn't. I need to add it manually.